src/connector/src/source/cdc/external/mysql.rs

```diff
@@ -115,65 +115,17 @@ impl MySqlExternalTable {
             // column name in mysql is case-insensitive, convert to lowercase
             let col_name = col.name.to_lowercase();
             let column_desc = if let Some(default) = col.default {
-                let snapshot_value = match default {
-                    ColumnDefault::Null => None,
-                    ColumnDefault::Int(val) => match data_type {
-                        DataType::Int16 => Some(ScalarImpl::Int16(val as _)),
-                        DataType::Int32 => Some(ScalarImpl::Int32(val as _)),
-                        DataType::Int64 => Some(ScalarImpl::Int64(val)),
-                        DataType::Varchar => {
-                            // should be the Enum type which is mapped to Varchar
-                            Some(ScalarImpl::from(val.to_string()))
-                        }
-                        _ => {
-                            tracing::error!(
-                                column = col_name,
-                                ?data_type,
-                                default_val = val,
-                                "unexpected default value type for column, set default to null"
-                            );
-                            None
-                        }
-                    },
-                    ColumnDefault::Real(val) => match data_type {
-                        DataType::Float32 => Some(ScalarImpl::Float32(F32::from(val as f32))),
-                        DataType::Float64 => Some(ScalarImpl::Float64(val.into())),
-                        DataType::Decimal => Some(ScalarImpl::Decimal(
-                            Decimal::try_from(val).map_err(|err| {
-                                anyhow!("failed to convert default value to decimal").context(err)
-                            })?,
-                        )),
-                        _ => {
-                            tracing::error!(
-                                column = col_name,
-                                ?data_type,
-                                default_val = val,
-                                "unexpected default value type for column, set default to null"
-                            );
-                            None
-                        }
-                    },
-                    ColumnDefault::String(mut val) => {
-                        // mysql timestamp is mapped to timestamptz, we use UTC timezone to
-                        // interpret its value
-                        if data_type == DataType::Timestamptz {
-                            val = timestamp_val_to_timestamptz(val.as_str())?;
-                        }
```

Comment:
Previously the implementation tried to handle all errors and fallback to `NULL`, but missed this line.

In this PR, we refactor the process into a separate function and handle errors through a unified path.